### PR TITLE
restructring examples

### PR DESCRIPTION
**Pull request by cagery**
- Proposes restructuring examples to categories.
- Adds related readmes.
- Removes "inference" word from folder names to simplify.

### CODE REVIEW

File: examples/nlp/README.md [hunk 1]
```diff
@@ -0,0 +1,3 @@
+# NLP Inference Examples
```
Comment (scxiao) [inline examples/nlp/README.md:1]: This can be written as Natural Language Processing (NLP) Inference Examples

File: examples/migraphx/README.md [hunk 2]
```diff
@@ -0,0 +1,7 @@
+# AMD MIGraphX usage and utilities
+
+- [C++ Parse, Load, and Save Graph Programs](./cpp_parse_load_save)
```
Comment (scxiao) [inline examples/migraphx/README.md:3]: For the readme file, can we mention "cmake .. -DCMAKE_PREFIX_PATH="migraphx_install_loc"?

File: examples/README.md [hunk 3]
```diff
@@ -1,18 +1,9 @@
...
-- [Exporting Frozen Graphs in TF2](./export_frozen_graph_tf2)
-- [MIGraphX Docker Container](./migraphx_docker)
-- [MIGraphX Driver](./migraphx_driver)
-- [Python Resnet50](./python_api_inference)
-- [Python BERT-SQuAD](./python_bert_squad_example)
-- [Python Super Resolution](./python_super_resolution)
-- [Python NFNet](./python_nfnet_inference)
-- [Python U-Net](./python_unet)
-- [Python 3D-UNet](./python_3dunet)
+- MIGraphX usage and utilities: [migraphx](./migraphx)
```
[the remark this answers is not in the document]
Comment (kahmed10) [reply examples/README.md:7]: Agree with Shucai above, but I think we should also include at least one sentence explaining what can be found in the link. So something like: [MIGraphX Usage and Utilities](./migraphx): description...

[the remark this answers is not in the document]
Comment (cagery) [reply examples/README.md]: Title change is done. I don't think the description is necessary, folder names and each folder's own description are self-explanatory.

File: examples/nlp/README.md (hunk 1 above)
Comment (cagery) [reply examples/nlp/README.md]: done

File: examples/README.md [hunk 4]
```diff
@@ -1,18 +1,9 @@
...
+- Vision inference examples: [vision](./vision)
```
[the remark this answers is not in the document]
Comment (cagery) [reply examples/README.md]: done

File: examples/migraphx/README.md (hunk 2 above)
Comment (cagery) [reply examples/migraphx/README.md:3]: I think this is covered in main readme file of migraphx, and each app in /migraphx/ examples has their own explanation.

File: examples/README.md [hunk 5]
```diff
@@ -4,6 +4,6 @@
 This directory contains examples of common use cases for MIGraphX. 
 
 ## Examples:
-- MIGraphX usage and utilities: [migraphx](./migraphx)
-- Vision inference examples: [vision](./vision)
-- Natural language inference examples: [nlp](./nlp)
\ No newline at end of file
+- MIGraphX usage and utilities: [MIGraphX usage and utilities](./migraphx)
+- Vision inference examples: [Vision inference examples](./vision)
+- Natural language inference examples: [Natural language inference examples](./nlp)
```
Comment (scxiao) [inline examples/README.md:9]: You can just change them to (no need for the duplication):

- [MIGraphX usage and utilities](./migraphx)
- [Vision inference examples](./vision)
- [Natural language inference examples](./nlp)

**Review by kahmed10 — APPROVED** [review]

File: examples/README.md (hunk 5 above)
Comment (cagery) [reply examples/README.md]: done

**Review by scxiao — APPROVED** [review]

**Review by turneram — APPROVED** [review]

**Pull request merged**